gson/src/main/java/com/google/gson/internal/bind/TypeAdapters.java

```diff
@@ -391,7 +392,15 @@ public Character read(JsonReader in) throws IOException {
       }
       String str = in.nextString();
       if (str.length() != 1) {
-        throw new JsonSyntaxException("Expecting character, got: " + str);
+        // maybe the input string is a number representing a char (like 102 for 'f')
+        try {
+          // so try returning a Character from its value, 
+          // first make in a real int, because JsonReader.nextString() returns String
+          return Character.valueOf((char) Integer.parseInt(str)); 
+        } catch (IllegalFormatException ex) {
```

Comment:
In case this pull request is still relevant: This should probably catch `NumberFormatException` not `IllegalFormatException`.

I am also not too sure whether the behavior of the default adapter should be changed. Maybe it would be better if users just registered their own adapter which has this behavior, if desired. Though I am not a member of this project and they might think differently about this.